sheetstorm/management/commands/data_load.py

```diff
@@ -0,0 +1,71 @@
+import logging
+
+from django.contrib.auth import get_user_model
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from users.factories import UserFactory
+
+CustomUser = get_user_model()
+
+user_type_options = {"admin": "admin", "manager": "manager", "employee": "employee"}
+
+password_for_every_user = "userpasswd"
+
+
+class Command(BaseCommand):
+    help = "Create initial sample data for SheetStorm application testing."
+
+    def add_arguments(self, parser):
+
+        parser.add_argument(
+            "-a", "--admin", type=int, help="Indicates the maximum number of admins to be in the database"
+        )
+        parser.add_argument(
+            "-e", "--employee", type=int, help="Indicates the maximum number of employees to be in the database"
+        )
+        parser.add_argument(
+            "-m", "--manager", type=int, help="Indicates the maximum number of managers to be in the database"
+        )
+
+    @staticmethod
+    def add_user(user_type, number_of_users_to_create):
+        if user_type == user_type_options["admin"]:
+            factory_parameters = {"user_type": CustomUser.UserType.ADMIN.name, "is_staff": True}
+
+        elif user_type == user_type_options["employee"]:
+            factory_parameters = {"user_type": CustomUser.UserType.EMPLOYEE.name}
+
+        elif user_type == user_type_options["manager"]:
+            factory_parameters = {"user_type": CustomUser.UserType.MANAGER.name}
+
+        else:
+            factory_parameters = {"user_type": CustomUser.UserType.ADMIN.name, "is_superuser": True, "is_staff": True}
+        for user_number in range(number_of_users_to_create):
+            user_email = f"user.{user_type}{user_number}@codepoets.it"
+            user = CustomUser.objects.filter(email=user_email).first()
+            if user is None:
+                user = UserFactory(**factory_parameters, email=user_email)
+                user.set_password(password_for_every_user)  # pylint: disable=unused-argument, no-value-for-parameter
+                user.save()
+                logging.info(f"Add {user_type}")
+
+    @transaction.atomic
+    def handle(self, *args, **options):
+
+        self.add_user("superuser", 1)
```

Comment:
Ok, so I find this thing problematic. You're giving him a "superuser" value here, but it's not checked anywhere inside the function -> Therefore you might as well type "orangutan" and it will still create a superuser. I don't think that's a good thing at all.

I'd suggest making **user_type** an optional argument and just call `self.add_user(1)`. Now there's only a question whether it makes conceptual and logical sense to have function **add_user** add superuser by default?

You can also add another if statement for "superuser" instead of else, but this brings us back to the assignment issue for **factory_parameters** ☹️ In this case, you could move this whole part:
```
        for user_number in range(number_of_users_to_create):
            user_email = f"user.{user_type}{user_number}@codepoets.it"
            user = CustomUser.objects.filter(email=user_email).first()
            if user is None:
                user = UserFactory(**factory_parameters, email=user_email)
                user.set_password(password_for_every_user)  # pylint: disable=unused-argument, no-value-for-parameter
                user.save()
                logging.info(f"Add {user_type}")
```
To a separate function. Like `create_and_add_user_to_database(factory_parameters, number_of_users_to_create)` for distinction. Then call them **under each if statement**, where **factory_parameters** is actually assigned. Therefore that problem will be solved.

I'll be happy with either approach, unless you'll think of something better, because then I will be overjoyed 😉